Support providing additional directories to copy

AverageMarcus opened the pull request:
Changes:

* Switched to using cobra for CLI to leverage the nice flag features
* Generate a new `go.mod` if not present in target directory (such as when initially generating a project)
* Support providing additional directories to copy over from the source. (e.g. the `exp` dir from CAPA)
* Support providing exclude globs to filter out unwanted files. (e.g. ignore all `_test.go` files from the source - hopefully cutting down on some deps)

AverageMarcus committed:
Support providing additional directories to copy

Also includes some small refactoring and checks.

Signed-off-by: Marcus Noble <github@marcusnoble.co.uk>

nprokopic commented on clone.go:
```diff
@@ -61,58 +62,56 @@ func Clone(c Config) error {
 
+	if _, err := os.Stat(filepath.Join(config.TargetDir, "go.mod")); os.IsNotExist(err) {
+		log.Println("No go.mod found in target directory, copying from source")
+		source, err := srcFilesystem.Open(filepath.Join(srcRoot, "go.mod"))
```
Not having original go.mod was basically one of the main motivations for this project 😀

The idea is to bring as little dependencies as possible from upstream project, because the go dependency tree from upstream projects has been causing issues for us (breaking our projects).

AverageMarcus replied:
Good point.

AverageMarcus replied:
Updated to instead generate a new one based on the module and go version of the source

nprokopic commented on config.go:
```diff
@@ -1,11 +1,12 @@
+	Repo           string
+	Tag            string
+	TargetDir      string
+	AdditionalDirs []string
```
What additional dirs you had in mind?

Even what it got copied in my tests (current wip branch of cluster-api-gen) is too much, as the idea is to get only the minimum number of packages required to get `api`.

AverageMarcus replied:
e.g.
```
goclone --org kubernetes-sigs --repo cluster-api-provider-aws --tag v1.0.0 --target-dir ./out --additional-dir exp/api`,
```

AverageMarcus committed:
Generate new go.mod

Signed-off-by: Marcus Noble <github@marcusnoble.co.uk>

AverageMarcus committed:
Added exclude flag

Supports providing globs to use to ignore copying files from source to destination.

Signed-off-by: Marcus Noble <github@marcusnoble.co.uk>
